File: wallet-app-backend/WalletApp.Application/Common/DefaultTransaction/EditDefaultUserTransaction/EditDefaultUserTransactionCommand.cs
using WalletApp.Application.Abstractions.Messaging;

namespace WalletApp.Application.Common.DefaultTransaction.EditDefaultUserTransaction;

public sealed record EditDefaultUserTransactionCommand(
    Guid Id,
    string Title,
    decimal Price,
    Guid CurrencyId,
    Guid CategoryId,
    string TextColor,
    string BackgroundColor,
    string Description
) : ICommand;

File: wallet-app-ui/src/models/apiTypes/defaultTransaction/EditDefaultUserTransaction/editDefaultUserTransaction.command.ts
export class EditDefaultUserTransactionCommand {
  constructor(
    public id: string,
    public title: string,
    public price: number,
    public currencyId: string,
    public categoryId: string,
    public textColor: string,
    public backgroundColor: string,
    public description?: string,
  ) {}
}
